### PR TITLE
feature/threat-scan

### PR DESCRIPTION
**Pull request by julianoborba**
Setup initial implementation of threat scan using X9Containers

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -5,12 +5,24 @@ inputs:
...
   repository_uri:
-    description: 'The URI of the ECR registry i.e. aws_account_id.dkr.ecr.region.amazonaws.com.'
+    description: The URI of the ECR registry i.e. aws_account_id.dkr.ecr.region.amazonaws.com.
 runs:
-  using: 'node12'
-  main: 'actionHandler.js'
-  post: 'cleanup.js'
+  using: "node12"
+  main: "actionHandler.js"
+  post: "cleanup.js"
```
Comment (carloosamarante) [inline action.yml:28]: on document they are using single quotes https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action

File: main.js [hunk 2]
```diff
@@ -120,7 +119,103 @@ const pushImage = async (config) => {
...
+  else if (`${config.minimalSeverity}` === 'HIGH')
+    minimalSeverity = 'HIGH,CRITICAL';
+  else if (`${config.minimalSeverity}` === 'MEDIUM')
+    minimalSeverity = 'MEDIUM,HIGH,CRITICAL';
+  else if (`${config.minimalSeverity}` === 'LOW')
+    minimalSeverity = 'LOW,MEDIUM,HIGH,CRITICAL';
+  else {
+    config.minimalSeverity = 'UNKNOWN';
+    minimalSeverity = 'UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL';
+  }
```
Comment (carloosamarante) [inline main.js:143]: here i think you could use a switch, is more readable, and a little bit faster, take a look https://medium.com/@michellekwong2/switch-vs-if-else-1d458e7b0711

File: main.js [hunk 3]
```diff
@@ -120,7 +119,103 @@ const pushImage = async (config) => {
...
+  if (grepClam.status !== 0) {
+    console.error(grepClam.stderr.toString());
+    return reject(`report image threats file ${clamScanFileName} grep failed`);
+  }
+  const summaryClam = grepClam.stdout.toString();
+  const totalsClam = summaryClam.match(/\d+/);
+  if (totalsClam.some(isNaN)) {
+    return reject(`report image threats file ${clamScanFileName} missing totals`);
+  }
+  process.stdout.write('ClamAV	');
```
Comment (carloosamarante) [inline main.js:181]: this blank space is on porpose?

File: action.yml (hunk 1 above)
Comment (julianoborba) [reply action.yml:28]: Visual Studio Code while auto-formating exchange to double quotes, but I will roll back that.

File: main.js (hunk 3 above)
Comment (carloosamarante) [reply main.js:181]: nice :)

File: main.js [hunk 4]
```diff
@@ -120,7 +119,109 @@ const pushImage = async (config) => {
...
+      break;
+    case 'LOW':
+      minimalSeverity = 'LOW,MEDIUM,HIGH,CRITICAL';
+      break;
+    default:
+      config.minimalSeverity = 'UNKNOWN';
+      minimalSeverity = 'UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL';
+      break;
+  }
+  const dockerBuild = spawnSync('docker', ['build', '-f', 'X9.Dockerfile', '-t', 'suspectimage', '--build-arg', `IMAGE=${ECR_ENDPOINT}/${config.repositoryNames[0]}:latest`, '--build-arg', `TRIVY_SEVERITY=${minimalSeverity}`, '--quiet', '.']);
```
Comment (marciorgb) [inline main.js:150]: This line is too big, can you  break it?

**Review by rogerfernandes — APPROVED** [review]

File: main.js (hunk 4 above)
Comment (marciorgb) [reply main.js:150]: Also, the line 125 has the same issue.

File: main.js [hunk 5]
```diff
@@ -1,40 +1,44 @@
...
 
 const AWS_ACCOUNT_ID = process.env.AWS_ACCOUNT_ID
 const ECR_ENDPOINT = `${AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com`
 
+const CRITICAL_THRESHOLD = 10;
+const HIGH_THRESHOLD = 250;
+const MEDIUM_THRESHOLD = 500;
+const LOW_THRESHOLD = 1000;
+const UNKNOWN_THRESHOLD = 200;
+const VIRUS_THRESHOLD = 0;
```
Comment (marciorgb) [inline main.js:21]: Can this lines be ordered by severity?

**Review by marciorgb — APPROVED** [review]

**Review by marciorgb — APPROVED** [review]

**Review by carloosamarante — APPROVED** [review]

**Review by rogerfernandes — APPROVED** [review]

**Pull request merged**